File: ims_coletor/IMS/servidor/servidor.py

# To change this template, choose Tools | Templates
# and open the template in the editor.
import commands


__author__="fabio"
__date__ ="$29/06/2009 15:21:14$"

class Servidor:
    id = None
    ip = None
    community = None
    servicos = list()
    
    def __init__(self, id, ip, community):
        self.id         = id
        self.ip         = ip
        self.community  = community
        
    def getIp(self):
        return self.ip

    def setIp(self, ip):
        self.ip = ip

    def getId(self):
        return self.id

    def setId(self, id):
        self.id = id

    def getCommunity(self):
        return self.community

    def setCommunity(self, community):
        self.community = community


    def getConsulta(self, mib):
        resultado = commands.getoutput("snmpget -v1 -c " + self.community + " " + self.ip + " " + mib + "|awk {'print $4'}")
        try:
        	resultado = resultado.split('"')[1]
        	return resultado
        except:
        	return resultado

    def getStatus(self):
        print commands.getoutput("snmpstatus -v1 -c " + self.community + " " + self.getIp())
         
    def getServicos(self):
        return self.servicos
    
    def setServicos(self, servicos):
        self.servicos.append(servicos)

File: src/java/br/fabio/model/servidor/Servidor.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package br.fabio.model.servidor;

import br.fabio.model.servico.Servico;
import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author fabio
 */
public class Servidor {

    private int codigo;
    private String nome;
    private String ip;
    private String hostname;
    private String community;
    private List<Servico> servicos;
    private String uptime;
    
    public void Servidor(){
        this.servicos = new ArrayList<Servico>();
    }

    public int getCodigo() {
        return codigo;
    }

    public String getNome() {
        return nome;
    }

    public String getIp() {
        return ip;
    }

    public String getHostname() {
        return hostname;
    }

    public String getCommunity() {
        return community;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public void setIp(String ip) {
        this.ip = ip;
    }

    public void setHostname(String hostname) {
        this.hostname = hostname;
    }

    public void setCommunity(String community) {
        this.community = community;
    }

    public List<Servico> getServicos() {
        return servicos;
    }

    public void setServicos(List<Servico> servicos) {
        this.servicos = servicos;
    }

    public void addServico(Servico servico) {
        this.servicos.add(servico);
    }

    public boolean temServico(int id){
        for (Servico servicoId : this.servicos){
            if (servicoId.getId() == id){
                return true;
            }
        }
        return false;
    }

    public String getUptime() {
        return uptime;
    }

    public void setUptime(String uptime) {
        this.uptime = uptime;
    }
    
}
